spacer/messages.py

```diff
@@ -41,74 +61,79 @@ def __hash__(self):
 
 
 class ExtractFeaturesMsg(DataClass):
-    """ Input message for extract features class. """
+    """ Input message for extract-features task. """
 
     def __init__(self,
                  job_token: str,  # Token for caller DB reference.
-                 feature_extractor_name: str,  # Which extractor to use.
-                 rowcols: List[Tuple[int, int]],  # List of [row, col] entries.
+                 extractor: 'FeatureExtractor',
+                 rowcols: list[tuple[int, int]],  # List of (row, col) entries.
                  image_loc: DataLocation,  # Where to fetch image.
                  feature_loc: DataLocation,  # Where to store output.
                  ):
 
-        assert feature_extractor_name in config.FEATURE_EXTRACTOR_NAMES
-        assert isinstance(rowcols, List)
+        assert isinstance(rowcols, list)
         assert len(rowcols) > 0, "Invalid message, rowcols entry is empty."
         assert len(rowcols[0]) == 2
-        assert feature_loc.storage_type != 'url', \
-            "Write not supported for url storage type."
+        assert feature_loc.is_writable, (
+            f"Write not supported for"
+            f" '{feature_loc.storage_type}' storage type.")
 
         self.job_token = job_token
-        self.feature_extractor_name = feature_extractor_name
+        self.extractor = extractor
         self.rowcols = rowcols
         self.image_loc = image_loc
         self.feature_loc = feature_loc
 
+    @classmethod
+    def example(cls) -> 'ExtractFeaturesMsg':
+        from spacer.extract_features import EfficientNetExtractor
```

Comment:
I'm probably just missing something but why not do the imports in the usual place up top?

Reply:
`extract_features.py` imports `messages`, so doing this method-level import avoids an unresolvable circular import situation.